src/components/Editable.tsx

```diff
@@ -269,6 +273,11 @@ const Editable = ({ disabled, isEditing, thoughtsRanked, contextChain, cursorOff
     dispatch(setEditingValue(newValue))
 
     if (newValue === oldValue) {
+      if (contentRef.current) {
+        contentRef.current.style.opacity = '1.0'
+      }
+      duplicateThoughtsAlertTimeout.current && window.clearTimeout(duplicateThoughtsAlertTimeout.current)
```

Comment:
We need to store timeout id somewhere (not in state) in order to clear timeout if user has changed duplicate thought value while alert about duplication is waiting to be dispatched in timeout

Reply:
Thanks! But why in a ref? Why not just use closure since they are only ever editing one thought at a time?

Reply:
We use this ref in different functions across Editable and its value could change (if during editing we detect 2+ duplication cases). I can write some utility function for handling duplication alert, define there timeoutId variable and return another function which invokes setTimeout and sets timeoutId, but it needs using let instead of const. Maybe i don't understand something about your idea? Correct me please, if i'm wrong somewhere.

Reply:
ref is used specifically to tie functionality to the mount/unmount cycle. I don't yet see how the duplicate functionality requires a ref. You can only edit one item at a time, so I don't see how 2+ duplicate cases are relevant.

A ref is more complicated than a simple `let` with closure, so I'm asking for you to provide justification for using this more complex structure. `const` is misleading because the ref has internal state that gets mutated, so it is no more functional than `let`. They both use state, albeit different kinds. Whenever cancelling a timeout is needed, state is involved.

What is stopping you from placing a variable with `let` at a high enough scope so that is accessible where it is needed? Is there a case that requires ref so that is bound to the mount/unmount cycle that I am not aware of? Again, I might be missing something, so let me know! Thanks!

Reply:
Yes, you are right. I didn't want to use let, but you made a good point, that ref and let are actually the same in term of mutating. I replaced ref with closure and was working on fixing this:
- a
  - ba
  - bac|

After hitting `Backspace` with the caret at the end of **bac** and then hitting `Enter`, alert is dispatched and doesn't hide after a while. I realized that hitting `Enter` triggers re-render component before onBlur and return function of useEffect are fired. After that closure is re-created and we lose timeout.